src/translators/sql_translator.py

```diff
@@ -1551,10 +1551,10 @@ def query_entity_types(self, fiware_service=None, fiware_servicepath=None):
             for et in all_types:
                 stmt = "select distinct(entity_type) from {}".format(et)
                 if fiware_servicepath == '/':
-                    stmt = stmt + " WHERE {} ~* '/.*'" \
+                    stmt = stmt + " WHERE {} = '/'" \
                         .format(FIWARE_SERVICEPATH)
                 elif fiware_servicepath:
-                    stmt = stmt + " WHERE {} ~* '{}($|/.*)'" \
+                    stmt = stmt + " WHERE {} = '{}'" \
```

Comment:
not sure about this changes

Reply:
most probably if your revert this, tests should pass

Reply:
I have removed ```.*``` as this results in extracting all the entities. I have also updated the if block as the statement can be used in both the cases.